File: common/src/main/java/de/werkelmann/rekote/address/ServerConnectionChecker.java
package de.werkelmann.rekote.address;

import java.util.LinkedList;
import java.util.List;

public abstract class ServerConnectionChecker {

    protected List<ConnectionRule> rules = new LinkedList<>();

    public boolean check(String address) {
        for (ConnectionRule rule : rules) {
            if (!rule.isValid(address)) {
                return false;
            }
        }
        return true;
    }
}

File: common/src/main/kotlin/de/werkelmann/rekote/address/ServerConnectionChecker.kt
package de.werkelmann.rekote.address

import java.util.*

abstract class ServerConnectionChecker {

    protected val rules: MutableList<ConnectionRule> = LinkedList()

    open fun check(address: String): Boolean {
        rules.forEach { if (!it.isValid(address)) return false }
        return true
    }
}
